e2e/jest.test-setup.ts

```diff
@@ -11,60 +8,87 @@ const isDebugMode = process.argv.includes('--debug');
  * - waitFor functions timeout
  */
 beforeEach(async () => {
-  await page.setUserAgent(userAgent);
-  await page.setViewport({width: 1280, height: 0});
-  page.setDefaultNavigationTimeout(60000); // Puppeteer default timeout is 30 seconds.
-  page.setDefaultTimeout(30000);
-});
 
-/**
- * At the end of each test completion, do:
- * - Disable network interception.
- * - Delete broswer cookies.
- * - Reset global page and browser variables.
- */
-afterEach(async () => {
-  await page.deleteCookie(...await page.cookies());
   await jestPuppeteer.resetPage();
   await jestPuppeteer.resetBrowser();
-});
 
-/**
- * Enable network interception in new page and block unwanted requests.
- */
-beforeAll(async () => {
+  await page.setUserAgent(userAgent);
+  await page.setViewport({width: 1300, height: 0});
+
+  page.setDefaultNavigationTimeout(60000); // Puppeteer default timeout is 30 seconds.
+  page.setDefaultTimeout(30000);
+
   await page.setRequestInterception(true);
+
   page.on('request', (request) => {
-    const requestUrl = url.parse(request.url(), true);
-    const host = requestUrl.hostname;
-    // to improve page load performance, block network requests unrelated to application.
     try {
-      if (host === 'www.google-analytics.com'
-           || host === 'accounts.youtube.com'
-           || host === 'static.zdassets.com'
-           || host === 'play.google.com'
-           || request.url().endsWith('content-security-index-report')) {
-        request.abort();
-      } else {
-        request.continue();
+      request.continue();
+      // tslint:disable-next-line:no-empty
+    } catch (e) {
+    }
+  });
+
+  page.on('console', async(message) => {
+    if (message != null) {
+      const msg = message.type().toUpperCase();
+      // Don't log "log", "info" or "debug"
+      if (msg.includes('ERROR') || msg.includes('WARNING')) {
+        console.debug(`❗ ${message.type()}: ${message.text()}`);
+      }
+    }
+  });
+
+  // Emitted when the page crashed
+  page.on('error', error => {
+    console.debug(`❗ ${error}`);
+  });
+
+  // Emitted when a script has uncaught exception
+  page.on('pageerror', error => {
+    if (error != null) {
+      console.debug(`❗ ${error}`);
```

Comment:
I think it's worth distinguishing between the different event types. We can add a prefix to each message to show that.